Clarifications of parameters in yum_repo module

AlexisLessard opened the pull request:
Added a note defining where the "name" parameter of the module will appear in the repo file and note explaining that the description parameter of the module is actually the name parameter in the repo file. It might help people transform their existing yum repository files in ansible managed repos.
+label: docsite_pr

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
yum_repository module

##### ANSIBLE VERSION
```
ansible 2.4.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.6/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.6.6 (r266:84292, Aug 18 2016, 15:13:37) [GCC 4.4.7 20120313 (Red Hat 4.4.7-17)]
```

AlexisLessard committed:
Clarifications of parameters in yum_repo module

Added a note defining where the "name" parameter of the module will appear in the repo file and note explaining that the description parameter of the module is actually the name parameter in the repo file. It might help people transform their existing yum repository files in ansible managed repos.
+label: docsite_pr

jtyr commented on lib/ansible/modules/packaging/os/yum_repository.py:
```diff
@@ -68,7 +68,7 @@
   description:
     description:
-      - A human readable string describing the repository.
+      - A human readable string describing the repository. Note that it corresponds to the "name" parameter in the yum repo file.
```
Let's use the terminology known for INI files:

```
This option corresponds to the "name" property in the repo file.
```

jtyr commented on lib/ansible/modules/packaging/os/yum_repository.py:
```diff
@@ -200,7 +200,7 @@
   name:
     description:
-      - Unique repository ID.
+      - Unique repository ID. In relation to the yum file, it's the ID of the section of the repository
```
The same like above:

```
This option builds the section name of the repository in the repo file.
```

AlexisLessard committed:
Corrections to respect ini terminology

AlexisLessard commented:
I specified in some places that we were talking about a .ini file, as I didn't know that yum repos were of this format, a format which I haven't worked a lot with. It's just a little clarifications that might give useful informations to ... well.. noobs like me 😄

jtyr commented on lib/ansible/modules/packaging/os/yum_repository.py:
```diff
@@ -200,7 +200,7 @@
   name:
     description:
-      - Unique repository ID.
+      - Unique repository ID. This option builds the section name of the repository in the .ini repo file.
```
I think it's not necessary to mention the `.ini` here.

AlexisLessard replied:
@jtyr , I disagree, as I didn't know that yum repos were defined by .ini files, and I'm not familiar with that format. I believe that some users might use ansible because they are unfamiliar with the manual use of repo files, thus helping them understood its utilization. What I mean is: I don't think just writing "section name" will help new users finding where the value will appear. That being said, if you disagree with me, I'll remove it.

AlexisLessard committed:
Removed .ini from the description 

I removed .ini from the description of the description attribute I previously specified. Users should be able to know .ini terminology.

acozine commented:
Thank you for contributing to the documentation @AlexisLessard, and congrats on your first merged PR!
